metadata-ingestion/tests/unit/test_unity_catalog_config.py

```diff
@@ -379,3 +379,119 @@ def test_ml_model_max_results_negative_invalid():
                 "ml_model_max_results": -100,
             }
         )
+
+
+def test_lineage_data_source_default():
+    """Test that lineage_data_source defaults to AUTO."""
+    config = UnityCatalogSourceConfig.parse_obj(
+        {
+            "token": "token",
+            "workspace_url": "https://test.databricks.com",
+            "include_hive_metastore": False,
+            "include_tags": False,
+        }
+    )
+    from datahub.ingestion.source.unity.config import LineageDataSource
+
+    assert config.lineage_data_source == LineageDataSource.AUTO
+
+
+def test_lineage_data_source_system_tables_requires_warehouse_id():
+    """Test that lineage_data_source=SYSTEM_TABLES requires warehouse_id."""
+    with pytest.raises(
+        ValueError,
+        match="lineage_data_source='SYSTEM_TABLES' requires warehouse_id to be set",
+    ):
+        UnityCatalogSourceConfig.parse_obj(
+            {
+                "token": "token",
+                "workspace_url": "https://test.databricks.com",
+                "include_hive_metastore": False,
+                "include_tags": False,
+                "lineage_data_source": "SYSTEM_TABLES",
+            }
+        )
+
+
+def test_lineage_data_source_api_does_not_require_warehouse():
+    """Test that lineage_data_source=API does not require warehouse_id."""
+    config = UnityCatalogSourceConfig.parse_obj(
+        {
+            "token": "token",
+            "workspace_url": "https://test.databricks.com",
+            "include_hive_metastore": False,
+            "include_tags": False,
+            "lineage_data_source": "API",
+        }
+    )
+    from datahub.ingestion.source.unity.config import LineageDataSource
+
+    assert config.lineage_data_source == LineageDataSource.API
+    assert config.warehouse_id is None
+
+
+def test_usage_data_source_default():
+    """Test that usage_data_source defaults to AUTO."""
+    config = UnityCatalogSourceConfig.parse_obj(
+        {
+            "token": "token",
+            "workspace_url": "https://test.databricks.com",
+            "include_hive_metastore": False,
+            "include_tags": False,
+        }
+    )
+    from datahub.ingestion.source.unity.config import UsageDataSource
+
+    assert config.usage_data_source == UsageDataSource.AUTO
+
+
+def test_usage_data_source_system_tables_requires_warehouse_id():
+    """Test that usage_data_source=SYSTEM_TABLES requires warehouse_id."""
```

Comment:
I wonder whether we should be adding such comments, do you find them helpful? Seems like they repeat what is already stated in the name of the test method.